server/src/main/java/org/apache/druid/server/http/DataSourcesResource.java

```diff
@@ -236,12 +250,24 @@ public Response markSegmentsAsUnused(
                    .collect(Collectors.toSet());
 
         // Note: segments for the "wrong" datasource are ignored.
-        return segmentsMetadataManager.markSegmentsAsUnused(
+        numUpdatedSegments = segmentsMetadataManager.markSegmentsAsUnused(
             segmentIds.stream()
                       .filter(segmentId -> segmentId.getDataSource().equals(dataSourceName))
                       .collect(Collectors.toSet())
         );
+        auditPayload = Collections.singletonMap("segmentIds", segmentIds);
+      }
+      if (auditManager != null && author != null && !author.isEmpty()) {
+        auditManager.doAudit(
+            AuditEvent.builder()
+                      .key(dataSourceName)
+                      .type("segments.markUnused")
```

Comment:
can the type name be standardized? E.g. serviceName.api-path

Reply:
Yeah, that makes sense to me.

So would something like the following work?
- `coordinator.markSegmentsUnused`
- `coordinator.basicAuth.createUser`
- `overlord.postTask`
- `overlord.killSegments`

Reply:
Added a field `request` in `AuditEntry` to contain the details of `path`, `uri`, `method` and `service`.